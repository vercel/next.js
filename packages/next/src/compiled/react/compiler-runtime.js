/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react-compiler-runtime.production.js');
} else {
  module.exports = require('./cjs/react-compiler-runtime.development.js');
}
