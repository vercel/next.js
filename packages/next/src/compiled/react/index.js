'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react.production.js');
} else {
  module.exports = require('./cjs/react.development.js');
}
