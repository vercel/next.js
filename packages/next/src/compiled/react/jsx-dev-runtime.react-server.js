'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react-jsx-dev-runtime.react-server.production.js');
} else {
  module.exports = require('./cjs/react-jsx-dev-runtime.react-server.development.js');
}
