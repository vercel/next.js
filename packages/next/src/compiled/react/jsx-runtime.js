'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react-jsx-runtime.production.js');
} else {
  module.exports = require('./cjs/react-jsx-runtime.development.js');
}
