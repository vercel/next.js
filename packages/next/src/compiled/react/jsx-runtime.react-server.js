'use strict';

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react-jsx-runtime.react-server.production.js');
} else {
  module.exports = require('./cjs/react-jsx-runtime.react-server.development.js');
}
