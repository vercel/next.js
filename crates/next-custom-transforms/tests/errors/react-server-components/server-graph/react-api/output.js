import { useState } from 'react';
import { createContext } from 'react';
import { useEffect, useImperativeHandle } from 'react';
import { Component, createFactory, PureComponent, useDeferredValue, useInsertionEffect, useLayoutEffect, useReducer, useRef, useSyncExternalStore } from 'react';
import { experimental_useOptimistic as useOptimistic } from 'react';
export default function() {
    return null;
}
