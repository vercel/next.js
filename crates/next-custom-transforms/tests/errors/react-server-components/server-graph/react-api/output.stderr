  x You're importing a component that needs `useState`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
   ,-[input.js:1:1]
 1 | import { useState } from 'react'
   :          ^^^^^^^^
   `----
  x You're importing a component that needs `createContext`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
   ,-[input.js:3:1]
 2 | 
 3 | import { createContext } from 'react'
   :          ^^^^^^^^^^^^^
   `----
  x You're importing a component that needs `useEffect`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
   ,-[input.js:5:1]
 4 | 
 5 | import { useEffect, useImperativeHandle } from 'react'
   :          ^^^^^^^^^
   `----
  x You're importing a component that needs `useImperativeHandle`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
   ,-[input.js:5:1]
 4 | 
 5 | import { useEffect, useImperativeHandle } from 'react'
   :                     ^^^^^^^^^^^^^^^^^^^
   `----
  x Youâ€™re importing a class component. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.
  | Learn more: https://nextjs.org/docs/app/building-your-application/rendering/client-components
  | 
  | 
   ,-[input.js:8:1]
 7 | import {
 8 |   Component,
   :   ^^^^^^^^^
 9 |   createFactory,
   `----
  x You're importing a component that needs `createFactory`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:9:1]
  8 |   Component,
  9 |   createFactory,
    :   ^^^^^^^^^^^^^
 10 |   PureComponent,
    `----
  x You're importing a component that needs `PureComponent`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:10:1]
  9 |   createFactory,
 10 |   PureComponent,
    :   ^^^^^^^^^^^^^
 11 |   useDeferredValue,
    `----
  x You're importing a component that needs `useDeferredValue`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:11:1]
 10 |   PureComponent,
 11 |   useDeferredValue,
    :   ^^^^^^^^^^^^^^^^
 12 |   useInsertionEffect,
    `----
  x You're importing a component that needs `useInsertionEffect`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:12:1]
 11 |   useDeferredValue,
 12 |   useInsertionEffect,
    :   ^^^^^^^^^^^^^^^^^^
 13 |   useLayoutEffect,
    `----
  x You're importing a component that needs `useLayoutEffect`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:13:1]
 12 |   useInsertionEffect,
 13 |   useLayoutEffect,
    :   ^^^^^^^^^^^^^^^
 14 |   useReducer,
    `----
  x You're importing a component that needs `useReducer`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:14:1]
 13 |   useLayoutEffect,
 14 |   useReducer,
    :   ^^^^^^^^^^
 15 |   useRef,
    `----
  x You're importing a component that needs `useRef`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:15:1]
 14 |   useReducer,
 15 |   useRef,
    :   ^^^^^^
 16 |   useSyncExternalStore,
    `----
  x You're importing a component that needs `useSyncExternalStore`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:16:1]
 15 |   useRef,
 16 |   useSyncExternalStore,
    :   ^^^^^^^^^^^^^^^^^^^^
 17 | } from 'react'
    `----
  x You're importing a component that needs `experimental_useOptimistic`. This React hook only works in a client component. To fix, mark the file (or its parent) with the `"use client"` directive.
  | 
  |  Learn more: https://nextjs.org/docs/app/api-reference/directives/use-client
  | 
  | 
    ,-[input.js:19:1]
 18 | 
 19 | import { experimental_useOptimistic as useOptimistic } from 'react'
    :          ^^^^^^^^^^^^^^^^^^^^^^^^^^
    `----
